app/models/megaphone/episode.rb

```diff
@@ -191,9 +212,31 @@ def audio_is_latest?
     end
 
     def set_enclosure
-      return unless audio_file_status == "success"
-      return if download_url.blank? || feeder_episode.enclosure_override_url.present?
-      feeder_episode.update(enclosure_override_url: download_url, enclosure_override_prefix: true)
+      return if download_url.blank? || !feeder_episode.complete_media?
+      set_external_media
+      feeder_episode.update!(enclosure_override_url: download_url, enclosure_override_prefix: true)
+    end
+
+    # if it's not published yet, can't get the file from mp, create as "complete"
+    def set_external_media
+      return if feeder_episode.published?
```

Comment:
if the episode is published, we can retrieve the info about the media, and verify it is there.

if it isn't, we rely on what we know the values will be, and by creating the external media, prevent it failing and staying "processing" until published.

There is some missing action, which is to validate the megaphone url on publish/release - which means waiting for the PR which tracks when an episode first shows up in the RSS.